### DISCUSSION TITLE
map refresh removes returned lines from external script

### DISCUSSION THREAD

**Discussion by chasdabigone** [scandum/tintin#233] [Script help]
Hi, I have a python script that I run using #system and it works when the visual map is not enabled. When the map is enabled, the returned lines disappear as soon as the map is refreshed (almost instantly). How can I make the lines continue to be rendered in the main tintin window?

**Comment by scandum** [comment]
Hi, you'll need to use either #run or #script. The messages generated by #system are external, so it's not something tintin is aware of.

**Reply by chasdabigone** [reply]
Even while using #run or #script it has the same behavior. The returned lines stay when the screen is '#split 0' but they are removed by refresh when the screen is ex. '#split 20 0'

**Reply by chasdabigone** [reply]
now I am really confused because I switched it back to #system and the lines are staying... maybe it is an issue with my terminal software

**Comment by scandum** [comment]
If you use `#script test ls;#var test` the data will stay when refreshing, but when using `#system ls` it won't.

**Reply by chasdabigone** [reply]
Ok, I can do it by storing variables and printing those. It seems to work with just returned lines to the text window as long as the map is not refreshing too quickly. I am pasting my sleuthing below, feel free to ignore if it is not something important.


The returned lines stay whether I use #system #run or #script if I am using the default #split.

When I do:
```
#split 0 1 0 -90
#map offset 1 91 -4 -1
```
at this point the lines returned by the python script are removed upon a map refresh, no matter if I use system, run, or script.

But if I then follow up with `#split 0` the map does not refresh immediately and the returned lines stay. The map will only refresh when I move in a direction. The lines will stay when the map refreshes though.

**Comment by scandum** [comment]
Tintin performs a full screen reprint of the scrollback buffer when in vertical split mode. So if your python script is sending messages to the terminal outside of tintin, they will indeed disappear. One easy work around is:

```
#line background #run bla bash
#snoop bla
#bla ls
```
